discojs/discojs-core/src/aggregator/base.ts

```diff
@@ -0,0 +1,239 @@
+import { client, Task, tf, AsyncInformant } from '..'
+
+import { EventEmitter } from 'events'
+
+import { Map, Set } from 'immutable'
+
+export enum AggregationStep {
+  ADD,
+  UPDATE,
+  AGGREGATE
+}
+
+export abstract class Base<T> {
+  /**
+   * Contains the ids of all active nodes, i.e. members of the aggregation group at
+   * a given round. It is a subset of all the nodes available in the network.
+   */
+  protected _nodes: Set<client.NodeID>
+  /**
+   * Contains the contributions received from active nodes, accessible
+   * by node id. It defines the effective aggregation group, which is
+   * possibly a subset of all active nodes, depending on the aggregation
+   * scheme.
+   */
+  protected contributions: Map<number, Map<client.NodeID, T>>
+  /**
+   * Emits the aggregation event whenever an aggregation step is performed.
+   * Triggers the resolve of the result promise and the preparation for the
+   * next aggregation round.
+   */
+  private readonly eventEmitter: EventEmitter
+
+  protected informant?: AsyncInformant<T>
+  /**
+   * The result promise which, on resolve, will contain the current aggregation result.
```

Comment:
want to say if it's a model (params) or model difference, or could be either?

Reply:
it can be anything really, since the class is generic, but in the case of the subclasses, it can either be the model weights or model weights difference

currently, I'm pretty sure the weights passed to the client are the entire model, which will change in the future polishing PR(s): write docs, support a directed communication graph, re-include DP, re-write byzantine-robustness, etc.)